lib/TvmClient.js

```diff
@@ -313,16 +314,18 @@ class TvmClient {
    * @param {object} options presign options
    * @param {string} options.blobName file path
    * @param {number} options.expiryInSeconds presign URL expiry duration
+   * @param {string} options.permissions premissions for presigned URL
    * @private
    * @returns {Promise<object>} credentials for service
    */
   async _getPresignCredentials (endpoint, options) {
-    if (!options || !options.expiryInSeconds || !options.blobName) {
-      logAndThrow(new codes.ERROR_MISSING_OPTION({ messageValues: ['expiryInSeconds or blobName'], sdkDetails: options }))
+    if (!options || !options.expiryInSeconds || !options.blobName || !options.permissions) {
+      logAndThrow(new codes.ERROR_MISSING_OPTION({ messageValues: ['missing one or more of required options blobName, expiryInSeconds and permissions'], sdkDetails: options }))
     }
+
     let fullUrl = urlJoin(this.apiUrl, endpoint, this.ow.namespace)
     fullUrl = fullUrl + '?' + EXPIRY_IN_SECONDS + '=' + options.expiryInSeconds +
-     '&' + BLOB_NAME + '=' + options.blobName
+     '&' + BLOB_NAME + '=' + options.blobName + '&' + PERMISSIONS + '=' + options.permissions
```

Comment:
The URL class handles some of this string manipulation for us, its less error prone ... 
```js
let newUrl = new URL(urlJoin(apiUrl, endpoint, namespace))
    newUrl.searchParams.append(EXPIRY_IN_SECONDS, options.expiryInSeconds )
    newUrl.searchParams.append(BLOB_NAME, options.blobName)
    newUrl.searchParams.append(PERMISSIONS, options.permissions)
```